lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else if (callback) {
+    throw new TypeError('callback must be a function');
```

Comment:
@thefourtheye so you are suggesting that this Type check shouldn't be done.

@trott suggested otherwise in --> https://github.com/nodejs/node/pull/3539#issuecomment-151330286